server/src/main/java/io/deephaven/server/session/SessionServiceGrpcImpl.java

```diff
@@ -216,14 +216,11 @@ public void publishFromTicket(
 
             Ticket resultId = request.getResultId();
 
-            ticketRouter.publish(session, resultId, "resultId", () -> {
-                // when publish is complete, complete the gRPC request
-                GrpcUtil.safelyComplete(responseObserver, PublishResponse.getDefaultInstance());
-            })
-                    .queryPerformanceRecorder(queryPerformanceRecorder)
-                    .require(source)
-                    .onError(responseObserver)
-                    .submit(source::get);
+            ticketRouter.publish(session, resultId, "resultId",
+                    SessionState.toErrorHandler(sre -> GrpcUtil.safelyError(responseObserver, sre)), source);
+
+            // Now that destination ticket is now resolvable, we can safely complete the response.
+            GrpcUtil.safelyComplete(responseObserver, PublishResponse.getDefaultInstance());
```

Comment:
I think this is incorrect; we need to hook up onPublish callback before we send complete. I believe this is the cause of the non-deterministic CI failure.